### PR TITLE
Add Flask Backend

### PR DESCRIPTION
**Pull request by nimwijetunga**
Added Docker for optional local development

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+click==6.7
```
Comment (imruahmed) [inline backend/requirements.txt]: can you take a look into using `pipenv` for a virtual environment

File: backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.6
```
Comment (imruahmed) [inline backend/Dockerfile]: can you add docs on how to setup/use docker

File: backend/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+BUILD Project from root directory (dancefest-web) before starting backend server using the following command:
+```
+$ npm run build
+```
+
+To create and enter the virtual enviroment type the following commands:
+```
+$ pipenv --python 3.5
+$ pipenv install --dev
```
Comment (imruahmed) [inline backend/README.md]: Do these need to be run every time?

File: backend/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+$ pipenv shell
+```
+
+Run the following commands to initalize env vars:
+```
+export FLASK_APP=server.py
+export PROJECT_NAME="dancefest-backend"
```
Comment (imruahmed) [inline backend/README.md]: Do these need to be run every time?

Comment (imruahmed) [reply backend/README.md]: Can we add them to the `.env`?

File: backend/api_routes.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from flask import Blueprint
+from flask import jsonify
```
Comment (imruahmed) [inline backend/api_routes.py]: unused

File: backend/README.md (hunk 4 above)
Comment (nimwijetunga) [reply backend/README.md]: I don't usually like putting a .env file in the git repo but if it's fine with you I can add it

**Pull request merged**